src/utilities/Api.js

```diff
@@ -0,0 +1,62 @@
+const API_URL = import.meta.env.VITE_API_URL;
+
+// fetch for categories
+export const getCategories = async () => {
+  const response = await fetch(`${API_URL}/categories"`);
+  return response.json();
+};
+
+export const addCategory = async (newCategory) => {
+  const response = await fetch(`${API_URL}/categories"`, {
+    method: "POST",
+    headers: { "Content-Type": "application/json" },
+    body: JSON.stringify(newCategory),
+  });
+  return response.json();
+};
+
+export const editCategory = async (categoryId, categoryEdit) => {
+  const response = await fetch(`${API_URL}/categories/${categoryId}`, {
+    method: "PATCH",
+    headers: { "Content-Type": "application/json" },
+    body: JSON.stringify(categoryId, categoryEdit),
```

Comment:
In here you are sending two objects, you need you need to send an Category object. (TS)